clang/include/clang/CConv/Utils.h

```diff
@@ -32,7 +32,44 @@ typedef std::map<PersistentSourceLoc,
 // Maps a Decl to the DeclStmt that defines the Decl.
 typedef std::map<clang::Decl *, clang::DeclStmt *> VariableDecltoStmtMap;
 
-
+// Replacement for boost:bimap. A wrapper class around two std::maps to enable
+// map lookup in from key to value or from value to key.
+template <typename KeyT, typename ValueT>
+class BiMap {
+public:
+  BiMap() { }
+  ~BiMap() { KtoVal.clear(); ValToK.clear();}
+
+  void insert(KeyT KO, ValueT VO) {
+    KtoVal[KO] = VO;
+    ValToK[VO] = KO;
+  }
+
+  typename std::map<KeyT, ValueT>::const_iterator find(const KeyT &K) {
+    return KtoVal.find(K);
+  }
```

Comment:
Why not just insist the programmer call `.left().find(...)` rather than provide `.find(...)` directly? I wonder if this could lead to confusion.

Reply:
This is supported by `boost::bimap` and was used in one or two places. I don't object to removing it.